packages/modular-scripts/src/__tests__/package.test.ts

```diff
@@ -0,0 +1,81 @@
+import tree from 'tree-view-for-tests';
+import execa from 'execa';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import getModularRoot from '../utils/getModularRoot';
+
+// this can take a while...
+jest.setTimeout(3 * 60 * 1000);
+
+function modular(str: string, opts: Record<string, unknown> = {}) {
+  return execa('yarnpkg', ['modular', ...str.split(' ')], {
+    cwd: getModularRoot(),
+    cleanup: true,
+    stdio: 'inherit',
+    ...opts,
+  });
+}
+
+describe('WHEN building a package with dynamic imports', () => {
+  beforeAll(async () => {
+    await modular('build tests-dynamic-imports --preserve-modules --private');
+  });
+
+  it('SHOULD then create a tree with the correct structure', () => {
+    expect(tree(path.join(getModularRoot(), 'dist', 'tests-dynamic-imports')))
+      .toMatchInlineSnapshot(`
+      "tests-dynamic-imports
+      ├─ README.md #1jv3l2q
+      ├─ dist-cjs
+      │  ├─ _virtual
+      │  │  ├─ _rollupPluginBabelHelpers.js #8zp60f
+      │  │  └─ _rollupPluginBabelHelpers.js.map #1a6wdsa
+      │  ├─ dynamic-index.js #k4zvgk
+      │  ├─ dynamic-index.js.map #1shgnup
+      │  ├─ index.js #1jdsfb4
+      │  └─ index.js.map #19flky6
+      ├─ dist-es
+      │  ├─ _virtual
+      │  │  ├─ _rollupPluginBabelHelpers.js #14tvdhd
+      │  │  └─ _rollupPluginBabelHelpers.js.map #4hotzf
```

Comment:
What's up with this nested virtual folder?

Reply:
It's generated via `@babel/preset-env` for generated helper functions which are used in the output.